crates/weaver_live_check/src/sample_metric.rs

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Intermediary format for telemetry sample spans
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use weaver_checker::violation::{Advice, AdviceLevel};
+use weaver_semconv::group::InstrumentSpec;
+
+use crate::{
+    live_checker::LiveChecker, sample_attribute::SampleAttribute, Error, LiveCheckResult,
+    LiveCheckRunner, LiveCheckStatistics, SampleRef, MISSING_METRIC_ADVICE_TYPE,
+};
+
+/// The data point types of a metric
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum DataPoints {
+    /// Number data points
+    Number(Vec<SampleNumberDataPoint>),
+    /// Histogram data points
+    Histogram(Vec<SampleHistogramDataPoint>),
+    /// Exponential histogram data points
+    ExponentialHistogram(Vec<SampleExponentialHistogramDataPoint>),
+    /// Summary data points
+    Summary(Vec<SampleSummaryDataPoint>),
+}
+
+/// Represents a single data point of a metric
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct SampleNumberDataPoint {
```

Comment:
I may be missing something, but I think we need the metric available when we live-check sample points.  For example, checking if the type of point is int vs. double or checking if attributes match required/recommended (or are missing) can only be done here.